full-service/src/service/gift_code.rs

```diff
@@ -459,7 +459,10 @@ where
         )?;
 
         let account_key: AccountKey = mc_util_serial::decode(&from_account.account_key)?;
-        let tx_proposal = unsigned_tx.sign(&account_key, fog_resolver)?;
+        let signer = LocalRingSigner::from(&account_key);
+        let mut rng = rand::thread_rng();
+        let tx = signing_data.sign(&signer, &mut rng)?;
+        let tx_proposal = TxProposal::new(tx, signing_data);
```

Comment:
I've seen this duplicated, is it possible to implement a method on `TxProposal` that does it?
`TxProposal::from_signing_data(&signing_data, &account_key)`?
Or maybe `TxProposal::new()` can be changed to do this?